app/src/main/java/com/kickstarter/viewmodels/CreatorDashboardViewModel.java

```diff
@@ -72,21 +73,25 @@ public ViewModel(final @NonNull Environment environment) {
 
       this.projectsForBottomSheet = projects;
 
-      this.latestProject = latestProject;
+      projects.map(ListUtils::first).subscribe(this.currentProject::onNext);
 
-      this.projectAndStats = latestProject
+      this.projectAndStats = currentProject
         .compose(combineLatestPair(projectStatsEnvelope));
 
-      this.startProjectActivity = latestProject
+      this.projectSwitcherProjectClickOutput = this.projectSwitcherClicked;
+
+      this.startProjectActivity = currentProject
         .compose(takeWhen(this.projectViewClicked))
         .map(p -> Pair.create(p, RefTag.dashboard()));
     }
 
     private final PublishSubject<Void> projectViewClicked = PublishSubject.create();
+    private final PublishSubject<Project> projectSwitcherClicked = PublishSubject.create();
+    private final BehaviorSubject<Project> currentProject = BehaviorSubject.create();
```

Comment:
since `currentProject` is just a local variable used in the ViewModel constructor u can remove this and replace L76 with:

```java
final Observable<Project> currentProject = projects.map(ListUtils::first);
```

Reply:
yeah but we also need to call `onNext` from the [new input](https://github.com/kickstarter/android-oss/pull/156/files#diff-e77a405946e815e7f08dcae820a52984R112) so not sure that would work?  Not sure this logic is as clear as it could be though.  Like I'm making a round trip from new input -> new output -> new input -> update `currentProject`.

Reply:
ahhhh gotcha, i missed that the view model input `this.viewModel.refreshProject(project);` is invoked in the `projectSwitcherProjectClickOutput` call. seems like we should have just one `projectAndStats` output that handles the switching logic in the view model, we can pair on it!